doc/implementation_comparison.md

```diff
@@ -0,0 +1,198 @@
+# grpc-dotnet and Grpc.Core comparison
+
+This document summarizes the differences between the two available implementations of gRPC in C#.
+
+## Where both implementations are the same
+
+- 100% wire compatible and interoperable with each other and with other gRPC implementations
+- same API for invoking and handling RPC calls
+- basic functionality (streaming, metadata, deadline, cancellation, ...)
+- using same codegen tooling and MSBuild Integration (Grpc.Tools)
+- differences in setting up server and client are documented here: https://docs.microsoft.com/en-us/aspnet/core/grpc/migration?view=aspnetcore-3.0
```

Comment:
This would be a difference so it doesn't quite fit under the "Where both implementation are the same" section.